MetalCypher/MetalCypher.metal

```diff
@@ -11,33 +11,47 @@
 
 using namespace metal;
 
-kernel void bruteForce(constant byte const  * hash      [[buffer(BruteForceParameterHash)]],
-                       constant uint const  * input     [[buffer(BruteForceParameterInput)]],
-                       device   uint        * match     [[buffer(BruteForceParameterMatch)]],
+kernel void bruteForce(constant uint4 const * hash      [[buffer(BruteForceParameterHash)]],
+                       constant uint  const * input     [[buffer(BruteForceParameterInput)]],
+                       device   atomic_uint * match     [[buffer(BruteForceParameterMatch)]],
                        thread   uint          threadID  [[thread_position_in_grid]]) {
+  if (atomic_load_explicit(match, memory_order_relaxed) > 0) return;
+  
   thread byte password[MAX_PASSWORD_LENGTH];
-  uint passwordSize;
+  thread uint passwordSize;
   
   passwordFrom(*input + threadID, password, &passwordSize);
-  MD5 md5(password, passwordSize);
+  thread MD5 md5(password, passwordSize);
+
+  thread bool isNotAMatch = (hash->x ^ md5.output.x) | (hash->y ^ md5.output.y) | (hash->z ^ md5.output.z) | (hash->w ^ md5.output.w);
```

Comment:
If you want to avoid the weird camel casing you could call this `nonMatching`

Reply:
Done